index.js

```diff
@@ -33,11 +34,39 @@ Toolkit.run(async tools => {
     body: env.renderString(body, templateVariables),
     title: env.renderString(attributes.title, templateVariables)
   }
-
   tools.log.debug('Templates compiled', templated)
-  tools.log.info(`Creating new issue ${templated.title}`)
+
+  if (updateExisting) {
+    let existingIssue
+    tools.log.info(`Fetching issues with title "${templated.title}"`)
+    try {
+      const existingIssues = await tools.github.search.issuesAndPullRequests({
+        q: `is:open is:issue repo:${process.env.GITHUB_REPOSITORY} in:title ${attributes.title}`,
+        order: 'created'
+      })
+      existingIssue = existingIssues.data.items[0] // Get the first item with this title
```

Comment:
Juuuust in case the search API returns some records that have a title that doesn't exactly match the existing one:

```suggestion
      existingIssue = existingIssues.data.items.find(issue => issue.title === templated.title)
```